tests/mapper/test_adjacency_generators.py

```diff
@@ -110,3 +110,22 @@ def test_partial_powers(barbell, num_powers):
         for j in range(num_powers):
             print(i, j)
             assert np.allclose(partial_powers[0, j, :], actual_powers[j][i, :])
+
+
+def test_weighted(barbell):
```

Comment:
This is just a copy of test_partial_powers, with `to_adjacency_matrix(weighted=True)` (and similarly in `AdjacencyPowerGenerator`) instead of `to_adjacency_matrix`, and it seems like that's not quite right. Is there more to the algorithm than that?